### PR TITLE
Fix cadence so that it does not require optical flow cadence

### PR DESCRIPTION
**Pull request by rewbs**
This PR should be reviewed and merged after #53 and #54.

I have validated this PR causes no deviation from baseline with `deforum test-e2e`, except for the warp and glitch presets which both include non-deterministic 

`diffusion_cadence`>1 should not require `optical_flow_cadence`. This fixes the Classic-3D-Motion* presets which currently fail with:

```
2024-05-21 14:57:27.361 - deforum_e2e_test_helpers.py:168 - ERROR - Exception during test run.
Traceback (most recent call last):

  File "/home/rewbs/code/deforum-studio/venv/bin/deforum", line 8, in <module>
    sys.exit(start_deforum_cli())
    │   │    └ <function start_deforum_cli at 0x79037492f1c0>
    │   └ <function ExitHook.__init__.<locals>.exit at 0x790268182830>
    └ <module 'sys' (built-in)>

  File "/home/rewbs/code/deforum-studio/src/deforum/commands/deforum_cli.py", line 311, in start_deforum_cli
    run_e2e_test(options, extra_args)
    │            │        └ {'settings_file': '/home/rewbs/deforum_storage/presets/settings/Classic-3D-Motion-4.txt'}
    │            └ {'prompts': {'0': 'A solo delorean speeding on an ethereal highway through time jumps, like in the iconic movie back to the f...
    └ <function run_e2e_test at 0x79037492ef80>

> File "/home/rewbs/code/deforum-studio/src/deforum/commands/deforum_e2e_test_helpers.py", line 157, in run_e2e_test
    animation = deforum(**extra_args, **options)
                │         │             └ {'prompts': {'0': 'A solo delorean speeding on an ethereal highway through time jumps, like in the iconic movie back to the f...
                │         └ {'settings_file': '/home/rewbs/deforum_storage/presets/settings/Classic-3D-Motion-4.txt'}
                └ <deforum.pipelines.deforum_animation.pipeline_deforum_animation.DeforumAnimationPipeline object at 0x7903a1127130>

  File "/home/rewbs/code/deforum-studio/src/deforum/pipelines/deforum_animation/pipeline_deforum_animation.py", line 176, in __call__
    self.run_post_fn_list()
    │    └ <function DeforumAnimationPipeline.run_post_fn_list at 0x790270a16e60>
    └ <deforum.pipelines.deforum_animation.pipeline_deforum_animation.DeforumAnimationPipeline object at 0x7903a1127130>

  File "/home/rewbs/code/deforum-studio/src/deforum/pipelines/deforum_animation/pipeline_deforum_animation.py", line 564, in run_post_fn_list
    fn(self)
    │  └ <deforum.pipelines.deforum_animation.pipeline_deforum_animation.DeforumAnimationPipeline object at 0x7903a1127130>
    └ <function film_interpolate_cls at 0x790270a0ba30>

  File "/home/rewbs/code/deforum-studio/src/deforum/pipelines/deforum_animation/animation_helpers.py", line 1257, in film_interpolate_cls
    interpolated = interpolator(cls.images, film_in_between_frames_count)
                   │            │   │       └ 0
                   │            │   └ []
                   │            └ <deforum.pipelines.deforum_animation.pipeline_deforum_animation.DeforumAnimationPipeline object at 0x7903a1127130>
                   └ <deforum.generators.film_interp_generator.FILMInterpolator object at 0x7900a4898670>

  File "/home/rewbs/code/deforum-studio/src/deforum/generators/film_interp_generator.py", line 19, in __call__
    interpolated = [images[0]]
                    └ []

IndexError: list index out of range
```

### CODE REVIEW

File: src/deforum/pipelines/deforum_animation/pipeline_deforum_animation.py [hunk 1]
```diff
@@ -321,7 +321,7 @@ def setup(self, *args, **kwargs) -> None:
         self.shoot_fns.append(get_generation_params)
 
         self.gen.turbo_steps = self.gen.get('diffusion_cadence', 1)
-        if self.gen.turbo_steps > 1 and self.gen.optical_flow_cadence != 'None':
```
Comment (rewbs) [inline src/deforum/pipelines/deforum_animation/pipeline_deforum_animation.py:324]: This is the main logic change in this PR.

File: src/deforum/pipelines/deforum_animation/animation_helpers.py [hunk 2]
```diff
@@ -3,51 +3,62 @@
 import math
 import os
 import random
+
```
Comment (rewbs) [inline src/deforum/pipelines/deforum_animation/animation_helpers.py:6]: Import changes are just formatting changes for linter.

File: setup.py [hunk 3]
```diff
@@ -77,7 +77,6 @@
     'transformers==4.40.2',
     'omegaconf==2.3.0',
     'aiohttp==3.8.6',
-    'scipy==1.11.3',
```
Comment (rewbs) [inline setup.py:80]: Deps changes are just removals of duplicates.

File: src/deforum/pipelines/deforum_animation/animation_helpers.py [hunk 4]
```diff
@@ -921,7 +920,7 @@ def generate_interpolated_frames(cls):
                 if cls.gen.hybrid_motion in ['Affine', 'Perspective']:
                     if cls.gen.hybrid_motion_use_prev_img:
                         matrix = get_matrix_for_hybrid_motion_prev(tween_frame_idx - 1, (cls.gen.width, cls.gen.height), cls.gen.inputfiles,
-                                                                   prev_img, cls.gen.hybrid_motion)
```
Comment (rewbs) [inline src/deforum/pipelines/deforum_animation/animation_helpers.py:924]: prev_img was not defined here; I believe `cls.gen.prev_img` is the correct one to look up.

**Comment by deforum** [comment]
commit fails on a clean install
```
Traceback (most recent call last):
  File "/home/charmander/projects/deforum/src/deforum/generators/comfy_sd_generator.py", line 334, in load_model
    from custom_nodes.onediff_comfy_nodes._nodes import BasicBoosterExecutor
ModuleNotFoundError: No module named 'custom_nodes.onediff_comfy_nodes'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/charmander/projects/deforum/tests/test_animation_pipeline.py", line 19, in <module>
    pipeline(**args)
  File "/home/charmander/projects/deforum/src/deforum/pipelines/deforum_animation/pipeline_deforum_animation.py", line 168, in __call__
    self.run_shoot_fn_list()
  File "/home/charmander/projects/deforum/src/deforum/pipelines/deforum_animation/pipeline_deforum_animation.py", line 552, in run_shoot_fn_list
    fn(self)
  File "/home/charmander/projects/deforum/src/deforum/pipelines/deforum_animation/animation_helpers.py", line 741, in main_generate_with_cls
    cls.gen.image = cls.generate()
                    ^^^^^^^^^^^^^^
  File "/home/charmander/projects/deforum/src/deforum/pipelines/deforum_animation/pipeline_deforum_animation.py", line 692, in generate
    processed = self.generator(**gen_args)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/charmander/miniconda3/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/charmander/projects/deforum/src/deforum/generators/comfy_sd_generator.py", line 445, in __call__
    self.load_model()
  File "/home/charmander/projects/deforum/src/deforum/generators/comfy_sd_generator.py", line 344, in load_model
    logger.warn("NOT using onediff due to initialisation error. If you meant to, please check onediff custom nodes and their deps are correctly installed. To hid this message, set ENABLE_ONEDIFF=false.", exc_info=True)
    ^^^^^^^^^^^
AttributeError: 'Logger' object has no attribute 'warn'
Processing:   0%|    ```

related to onediff support

**Pull request merged**